cestzam-core/src/main/java/be/valuya/cestzam/core/service/myminfin/MyminfinServiceProvider.java

```diff
@@ -0,0 +1,45 @@
+package be.valuya.cestzam.core.service.myminfin;
+
+import be.valuya.cestzam.client.error.CestzamClientError;
+import be.valuya.cestzam.client.myminfin.rest.MyminfinRestClientService;
+import be.valuya.cestzam.client.myminfin.rest.UserData;
+import be.valuya.cestzam.core.service.CestzamServiceProvider;
+import be.valuya.cestzam.core.session.CestzamSession;
+import be.valuya.cestzam.core.session.cookie.CestzamCookies;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Locale;
+
+@ApplicationScoped
+public class MyminfinServiceProvider implements CestzamServiceProvider {
+
+    public static final String MYMINFIN = "MYMINFIN";
+
+    @Inject
+    MyminfinRestClientService restClientService;
+
+    @Override
+    public String getIdentifier() {
+        return MYMINFIN;
+    }
+
+    @Override
+    public CestzamSession initSession(Locale locale) {
+        CestzamCookies cookies = new CestzamCookies();
+        MyminfinSession myminfinSession = new MyminfinSession(MyminfinSessionStatus.UNAUTHENTICATED);
+        CestzamSession cestzamSession = new CestzamSession(cookies, locale, null, myminfinSession);
+
+        // TODO: Navigate to myminfin, click login
```

Comment:
the user initiate a login from myminfin

- click on locale select
- click on login

once authenticated, some users are required to choose the capacity (person / enterprise) and mandate they want to use